File: 1.arrays/quicksort/quick_sort.py
def quick_sort(arr):
  if not arr or len(arr) <=1:
    return arr
  mid = int((0 + len(arr)-1)/2)
  pivot = arr[mid]
  left = []
  right = []
  for i in range(0,len(arr)):
    if i == mid:
      continue
    if arr[i] <= pivot:
      left.append(arr[i])
    else:
      right.append(arr[i])
  return quick_sort(left) + [pivot] + quick_sort(right)


arr = [10, 80, 30, 90, 40, 50, 70]
print(quick_sort(arr))

File: 1.arrays/quicksort/quickSort.js
function quickSort(arr) {
  if (!arr || arr.length <= 1) {
    return arr;
  }
  const mid = Math.floor((0 + arr.length - 1) / 2);
  const pivot = arr[mid];
  const left = [];
  const right = [];
  for (let i = 0; i < arr.length; ++i) {
    if (i === mid) continue;
    if (arr[i] <= pivot) {
      left.push(arr[i]);
    } else {
      right.push(arr[i]);
    }
  }
  return [...quickSort(left), pivot, ...quickSort(right)];
}

const arr = [10, 80, 30, 90, 40, 50, 70];
console.log(quickSort(arr));
